x-pack/plugins/stack_connectors/server/connector_types/gen_ai/gen_ai.ts

```diff
@@ -73,4 +85,34 @@ export class GenAiConnector extends SubActionConnector<GenAiConfig, GenAiSecrets
     });
     return response.data;
   }
+
+  public async getDashboard({
+    dashboardId,
+  }: GenAiDashboardActionParams): Promise<GenAiDashboardActionResponse> {
+    const privilege = (await this.esClient.transport.request({
+      path: '/_security/user/_has_privileges',
+      method: 'POST',
+      body: {
+        index: [
+          {
+            names: ['*.kibana-event-log-*'],
```

Comment:
```suggestion
            names: ['.kibana-event-log-*'],
```

Reply:
> . However, I don't think that we should use a sub-action to do so

@kobelb Could you please provide an alternative solution? This is how API calls are made from other connectors such as Jira

Reply:
keep in mind that i am creating a saved object here, so I will need to do work server side. i really thought this was the cleanest way

Reply:
I anticipated this to be done using a dedicated internal HTTP route. All of the JIRA sub-actions are used to communicate with JIRA. Where-as, this PR is using a sub-action to create a Kibana Dashboard, not communicate with the Gen AI HTTP API.

Since this code is temporary, I don't have a strong opinion on whether we need to switch from the approach in this PR. I haven't been able to think of a strong concrete reason why using this approach temporarily would be an issue. However, it's not how we anticipated sub-actions to be used, or how they've been used elsewhere.

Reply:
The advantage to subactions is I can access `esClient` and `savedObjectsClient` from the actions plugin. If I built routes, I would need to pull more plugins into `stack_connectors`